libportal/remote.c

```diff
@@ -41,6 +42,68 @@ typedef struct {
   guint cancelled_id;
 } CreateCall;
 
+/**
+ * xdp_session_connect_to_eis
+ * @session: a [class@Session]
+ *
+ * Connect this XdpRemoteDesktopSession to an EIS implementation and return the fd.
+ * This fd can be passed into ei_setup_backend_fd(). See the libei
+ * documentation for details.
+ *
+ * This call must be issued before xdp_session_start(). If successful, all input
+ * event emulation must be handled via the EIS connection and calls to
+ * xdp_session_pointer_motion() etc. are silently ignored.
+ *
+ * Returns: the file descriptor to the EIS implementation or a negative errno
+ * on failure.
```

Comment:
Hm... returning `int`s is not a pattern we use in libportal. I think it would be better to return a `gboolean`, and add a outgoing GError argument to this function. Something like:

```c
gboolean
xdp_session_connect_to_eis (XdpSession  *session
                            GError     **error)
```

Reply:
done! in this case we do still need to return the `fd`, so I've changed this now to
```
int
xdp_session_connect_to_eis (XdpSession  *session
                            GError     **error)
```
and the return value is just `-1` in case of error. which roughly matches the existing
```
int xdp_session_open_pipewire_remote (XdpSession *session)
```
but that one doesn't (need to?) return a `GError`. I've simplified the error values a bit though so the caller either get's `NOT_SUPPORTED` or just `INVALID_VALUE`.